app/pages/events/activity_edit.php

```diff
@@ -3,14 +3,20 @@
 
 restrict_access(Access::$ADD_EVENTS);
 
-$event_id = get_route_param("event_id");
+$event_id = get_route_param("event_id", strict: false);
 $activity_id = get_route_param("activity_id", false);
-$event = em()->find(Event::class, $event_id);
-if (!$event) {
+$event = $event_id ? em()->find(Event::class, $event_id) : null;
+if ($event_id && !$event) {
     return "this event does not exist";
 }
 if ($activity_id) {
     $activity = em()->find(Activity::class, $activity_id);
+    if (!$activity) {
+        redirect($event_id ? "/evenements/$event_id/ajouter-activite" : "/ajouter-activite");
+    }
+    if ($activity->event->id != $event_id) {
+        redirect("/evenements/$activity->event->id/activite/$activity_id/modifier");
```

Comment:
This link might be broken
```suggestion
        redirect("/evenements/{$activity->event->id}/activite/$activity_id/modifier");
```

Reply:
Done